src/algebra/bit-manipulation.md

```diff
@@ -45,8 +45,10 @@ All those introduced operators are instant (same speed as an addition) on a CPU
 
 ### Bitwise operators
 
--   $\&$ : The bitwise AND operator compares each bit of its first operand with the corresponding bit of its second operand. 
-    If both bits are 1, the corresponding result bit is set to 1. Otherwise, the corresponding result bit is set to 0.
+-   & : The bitwise AND operator compares each bit of its first operand with the corresponding bit of its second operand. 
+    If both bits are 1, the corresponding result bit is set to 1. Otherwise, the corresponding result bit is set to 0.Using the and operation, we can check if a number x is even because x & 1 = 0
+    if x is even, and x & 1 = 1 if x is odd. More generally, x is divisible by 2k exactly
+    when x & (2k − 1) = 0.
```

Comment:
That trick doesn't really fit here. This small enumeration just shows the basic functions of the few operators, but no tricks.

You could make a new subsection, e.g. after the subsection "Check if a bit is set" called "Check if the number is divisible by a power of two" and explain it there.

Also make sure that you get the format correct, we use Markdown for the text, and Latex for the math formulas. Check the remaining document for how they are used.
And you can use https://cp-algorithms.com/preview.html for testing if your changes display correctly or not.

E.g. at the moment your explanations looks like you want to show a trick for divisibility by $2k$, but you actually want to show divisibility by $2^k$.